### PR TITLE
flux-queue: rewrite in python

### PR DESCRIPTION
**Pull request by chu11**
Per discussion in #4763 / #4885, lets start by re-writing `flux-queue` in python before moving onto more things.

This is pretty much a direct translation into Python with python-izing a bunch of code style.

Maybe debate could be on the error or usage output?  Interface and output should be identical.

### CODE REVIEW

**Comment by chu11** [comment]
re-pushed, squashing some fixups, had to battle finding something the github-code-scanning and CI python linters liked

**Comment by chu11** [comment]
hopefully no one started to review, re-pushed, found 2 nits and decided to tweak some code to be a little more pythonic

File: t/t2240-queue-cmd.t [hunk 1]
```diff
@@ -11,37 +11,37 @@ LIST_JOBS=${FLUX_BUILD_DIR}/t/job-manager/list-jobs
 
 test_expect_success 'flux-queue: unknown sub-command fails with usage message' '
 	test_must_fail flux queue wrongsubcmd 2>usage.out &&
-	grep Usage: usage.out
+	grep usage: usage.out
```
Comment (grondo) [inline t/t2240-queue-cmd.t:14]: Could `grep -i` be used here and this test moved previous to the change in output to avoid a commit that breaks tests?

**Review by grondo — APPROVED** [review]
Nice job getting this done. Just a couple very minor comments inline.

File: t/t2240-queue-cmd.t (hunk 1 above)
Comment (chu11) [reply t/t2240-queue-cmd.t:14]: oh good idea!

**Comment by chu11** [comment]
re-pushed with fixes per comments above, will set MWP

**Pull request merged**